### PR TITLE
Feature/layout fixes albert

### PR DESCRIPTION
**Pull request by albertpeiro**

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -1,51 +1,101 @@
...
+    <div class="container my-4">
+      <h2>Pick a series</h2>
+      <div id="allSeries">
+        <b-list-group>
+          <b-list-group-item
+            v-on:click="resultsPane.displaySeriesData(series)"
+            href="#"
+            v-for="series in seriesList"
+          >
+            {{ series.split('_')[0] }} {{ series.split('_')[1] }}
```
Comment (NZJourneyMan) [inline templates/index.html]:
```suggestion
            {{ series.split('_').join(' ') }}
```

Comment (albertpeiro) [reply templates/index.html]: even better using .replace

Comment (albertpeiro) [reply templates/index.html]: changed it

File: static/dswc_leaderboard.js [hunk 2]
```diff
@@ -10,74 +9,93 @@ var dataSource = "";
...
+        var highlightResult =
+            result.place === 1
+                ? 'primary'
+                : result.place === 2 || result.place === 3
+                ? 'success'
+                : '';
+        dRec._cellVariants = { place: highlightResult };
+
+        for (var r in result.races) {
+            var raceStr = 'R' + parseInt(r, 10) + 1;
```
Comment (NZJourneyMan) [inline static/dswc_leaderboard.js]:
```suggestion
            var raceStr = 'R' + (parseInt(r) + 1);
```

Comment (NZJourneyMan) [reply static/dswc_leaderboard.js]: I hate dynamically typed languages sometimes...

File: static/dswc_leaderboard.js [hunk 3]
```diff
@@ -10,74 +9,93 @@ var dataSource = "";
...
-        /* {'key': 'boatNum', class: 'text-center'}, */ 
-        {'key': 'points', class: 'text-center'},
+        { key: 'crew', stickyColumn: true },
+        { key: 'place', class: 'text-center' },
+        /* {'key': 'boatNum', class: 'text-center'}, */
+
+        { key: 'points', class: 'text-center' }
     ];
-    for (var b in data) {
+    for (var result of data) {
```
Comment (NZJourneyMan) [inline static/dswc_leaderboard.js:26]: `result of` I *knew* there would be a way of doing this... :)

File: static/dswc_leaderboard.js [hunk 4]
```diff
@@ -10,74 +9,93 @@ var dataSource = "";
...
+            .then(response => (this.seriesList = response.data));
     }
 });
 
 Vue.component('result-table', {
     props: ['title', 'tabledata'],
     template: `
         <div class='result-table'>
-            <h3>{{ title }}</h3>
+            <h3>Results for <span class="text-primary">{{ title.split('_')[0] }} {{ title.split('_')[1] }}</span></h3>
```
Comment (NZJourneyMan) [inline static/dswc_leaderboard.js]:
```suggestion
            <h3>Results for <span class="text-primary">{{ title.replace('_', ' ') }}</span></h3>
```

I wonder if we should add a css element to capitalise the first letter of the title, can we use css to extend the text-primary class?:
```
span.resultsTitle {
  text-transform: capitalize;
  ...
}
```

File: templates/index.html [hunk 5]
```diff
@@ -1,51 +1,117 @@
...
+                    <div class="mb-4">
+                        <h2 class="lead">Pick a series</h2>
+                        <div id="allSeries">
+                            <b-list-group>
+                                <b-list-group-item
+                                    v-on:click="resultsPane.displaySeriesData(series)"
+                                    href="#"
+                                    v-for="series in seriesList"
+                                >
+                                    {{ series.replace('_', ' ') }}
```
Comment (NZJourneyMan) [inline templates/index.html:55]: We could add in the capitalise span in here as well.

File: templates/index.html [hunk 6]
```diff
@@ -1,51 +1,117 @@
...
+                    </div>
+                </div>
+                <div class="col-lg-9">
+                    <div id="resultsPane" v-show="showResults" class="mb-4">
+                        <result-table
+                            :title="summaryTitle"
+                            :tabledata="summaryData"
+                        ></result-table>
+                        <!-- <div v-for="races in raceDays">
+                        <result-table :title="summaryTitle" :tabledata="summaryData"></result-table>
```
Comment (NZJourneyMan) [inline templates/index.html:68]: The next bit I was going to add was to remove the above summary table, but use this commented out section and provide an array of results:
```
- Leaderboard
- Round 1
...
- Round n
```
How does that fit in with your design?

File: templates/index.html [hunk 7]
```diff
@@ -1,51 +1,117 @@
...
+                    </div> -->
+                    </div>
+
+                    <div class="mb-4">
+                        Legend: <span class="badge badge-info">Computed</span>
```
Comment (NZJourneyMan) [inline templates/index.html]:
```suggestion
                        Legend: <span class="badge badge-info">Race result</span>
```

File: templates/index.html [hunk 8]
```diff
@@ -1,51 +1,117 @@
...
+                        <span class="badge badge-warning"
+                            >Missed / No data available</span
```
Comment (NZJourneyMan) [inline templates/index.html]: This is a computed from DNS, DNC, DSQ or DNF. Maybe the badge could say one of:
- Computed  (my preference)
- DNS/DNC/DSQ/DNF

Some systems print the flag value under the computed place, but I think the grid looks too busy already

This badge should be last in the list

File: templates/index.html [hunk 9]
```diff
@@ -1,51 +1,117 @@
...
+                        >
+                        <span class="badge badge-danger">Discarded</span>
+                    </div>
+                    <div class="mb-4">
+                        <p class="lead">Questions &amp; Comments</p>
+                        <p>
+                            If data for your race is missing or you think
+                            there's a mistake please let us know on the DSWC
+                            WhatsApp group!
```
Comment (NZJourneyMan) [inline templates/index.html]: Can we shorten this? It looks a bit messy on my mobile. 

Maybe just say "If race data is wrong or missing let us know on the WhatsApp Group!"

**Review by NZJourneyMan — COMMENTED** [review]
Ok, I was trying different buttons. So we have a review now instead of piece by piece comments. Lets see what works best :)

File: static/dswc_leaderboard.js [hunk 10]
```diff
@@ -10,74 +9,93 @@ var dataSource = "";
...
-        for (var r in data[b].races) {
-            var raceStr = parseInt(r, 10) + 1;
-            raceStr = 'R' + raceStr;
+
+        /* dRec.boatNum = result.boatNum; */
+        dRec.place = result.place;
+        dRec.points = result.points;
+        dRec.crew =
+            (result.place === 1
+                ? '🥇'
```
Comment (NZJourneyMan) [inline static/dswc_leaderboard.js:34]: The medals aren't rendering on my Chrome browser (on linux), but are rendering on my phone

**Review by NZJourneyMan — APPROVED** [review]

**Pull request merged**